src/Schema/PostgreSQLSchemaManager.php

```diff
@@ -610,13 +610,50 @@ protected function selectTableNames(string $databaseName): Result
 
     protected function selectTableColumns(string $databaseName, ?string $tableName = null): Result
     {
+        
+        // feature detection: if attrgenerated exists it means we are 
+        // in a postgreSQL version supporting GENERATED columns
+        // see https://pgpedia.info/g/generated-column.html#implementation
+        $results = $this->_conn->executeQuery(<<<'SQL'
+            SELECT true
+            FROM information_schema.columns
+            WHERE table_name = 'pg_attribute'
+                AND column_name = 'attgenerated';
+        SQL);
+        $supportsGeneratedColumn = $results->fetchOne();
+
+        $fragmentDefaultValueSQL = <<<SQL
+            SELECT pg_get_expr(adbin, adrelid)
+             FROM pg_attrdef
+             WHERE c.oid = pg_attrdef.adrelid
+                AND pg_attrdef.adnum=a.attnum
+        SQL;
+        if ($supportsGeneratedColumn) {
+            // in case of GENERATED ALWAYS AS ( foobar) STORED column
+            // postgreSQL's pg_get_expr(adbin, adrelid) will return the 'foobar' part
+            // which is not the 'default' value of the column but its 'definition'
+            // so in that case we force it to NULL as DBAL will use that column only for the
+            // 'default' value
+            $fragmentDefaultValueSQL = <<<SQL
+                SELECT
+                    CASE
+                        WHEN a.attgenerated = 's' THEN NULL
+                        ELSE pg_get_expr(adbin, adrelid)
+                    END
+                 FROM pg_attrdef
+                 WHERE c.oid = pg_attrdef.adrelid
+                    AND pg_attrdef.adnum=a.attnum
+            SQL;
+        }
+
+        
         $sql = 'SELECT';
 
         if ($tableName === null) {
             $sql .= ' c.relname AS table_name, n.nspname AS schema_name,';
         }
 
-        $sql .= <<<'SQL'
+        $sql .= <<<SQL
```

Comment:
I've remod the `''` to be able to inject the  subquery

I know it's not good, but I didn't find a cleaner way

Reply:
This is fine.